frontend/packages/kubevirt-plugin/integration-tests/views/vm.environment.view.ts

```diff
@@ -0,0 +1,62 @@
+import { $, $$, browser, by, ExpectedConditions as until, element } from 'protractor';
+import { addVariableFrom } from '@console/internal-integration-tests/views/environment.view';
+import {
+  PAGE_LOAD_TIMEOUT_SECS,
+  VM_BOOTUP_TIMEOUT_SECS,
+  KUBEVIRT_SCRIPTS_PATH,
+} from '../tests/utils/consts';
+import { actionsButton } from '@console/internal-integration-tests/views/crud.view';
+import { execSync } from 'child_process';
+
+export const selectResource = element(by.buttonText('Select a resource'));
+export const allPairRows = $$('.pairs-list__row');
+export const dropDownfilter = $('[data-test-id="dropdown-text-filter"]');
+export const resouceNameInput = $('.co-resource-item__resource-name');
+export const clickFirstOption = async () => {
+  await $$('[role=option]')
+    .get(0)
+    .$('.pf-c-dropdown__menu-item')
+    .click();
+};
+export const saveButton = $('.pf-c-form__actions').$('.pf-m-primary');
+export const successAlert = $('.pf-c-alert.pf-m-inline.pf-m-success.co-alert');
+export const errorAlert = $('.pf-c-alert.pf-m-inline.pf-m-danger.co-alert.co-alert--scrollable');
+
+export const startVMChoice = $('[data-test-action="Start Virtual Machine"]');
+
+export const dropDownBtn = $$('.value-from');
+export const textFilter = $('[placeholder="Config Map or Secret"]');
+export const option = $$('[role="option"]');
+export const deleteButton = $$('.pf-c-button.pf-m-plain.pairs-list__span-btns');
+
+export const serialField = $$('[data-test-id=env-prefix]');
+export const saveBtn = element(by.cssContainingText('.pf-m-primary', 'Save'));
+
+export const diskTable = $('.co-m-table-grid--bordered');
+
+export const noSerialError = 'Some sources are missing a Serial Number';
+export const dupSerialsError = 'There are two or more sources with the same Serial Number';
+
+const addVariableButton = element(by.buttonText('Add All From Config Map or Secret'));
+
+export const clickAddSourceButton = async () => {
+  await browser.wait(until.elementToBeClickable(addVariableButton), PAGE_LOAD_TIMEOUT_SECS);
+  await element(by.buttonText('Add All From Config Map or Secret')).click();
+};
+
+export const addSource = async (sourceName) => {
+  await clickAddSourceButton();
+  await addVariableFrom(sourceName);
+};
+
+export const timeout = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const startVM = async (vmName: string, ns: string) => {
```

Comment:
I think we shouldn't re-implement this, it's better to create a VirtualMachine instance and use it's action method
`const vm = new VirtualMachine();`
`await vm.action(VM_ACTION.Start);`